src/markets.rs

```diff
@@ -0,0 +1,150 @@
+mod kraken;
+use chrono::{DateTime, Utc};
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash, strum_macros::Display)]
+pub enum TradingPair {
+    BtcDai,
+}
+
+#[derive(Debug, Copy, Clone, PartialEq)]
+pub struct Rate {
+    trading_pair: TradingPair,
+    position: Position,
+    rate: f64,
+    timestamp: DateTime<Utc>,
+}
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash, strum_macros::Display)]
+pub enum Position {
+    BUY,
+    SELL,
+}
+
+pub struct Ohlc {
+    high: f64,
+    low: f64,
+    vwap: f64,
+    timestamp: DateTime<Utc>,
+    trading_pair: TradingPair,
+}
+
+impl Ohlc {
+    fn to_rate(&self, position: Position) -> anyhow::Result<Rate> {
+        let rate = if self.vwap == 0.0 {
+            let precision = 0.000_000_000_000_001;
```

Comment:
Rust [f64 documentation](https://doc.rust-lang.org/std/primitive.f64.html) use `10e-10` as the max precision, I'd suggest to do the same.

Reply:
done